### PR TITLE
doc: Update vUART tutorial

### PR DESCRIPTION
**Pull request by amyreye**
- Update overview, dependencies and constraints
- Update to match Configurator UI instead of manually editing XML files
- Remove architectural details and instead point to high-level design documentation

Signed-off-by: Reyes, Amy <amy.reyes@intel.com>

### CODE REVIEW

**Review by NanlinXie — APPROVED** [review]

File: doc/tutorials/vuart_configuration.rst [hunk 1]
```diff
@@ -3,386 +3,151 @@
...
-|                 | (vUART enable)        | (vUART enable)     | RTVM           |                |
-+-----------------+-----------------------+--------------------+----------------+----------------+
-
-.. _how-to-configure-a-console-port:
+A virtual universal asynchronous receiver/transmitter (vUART) can be a console
+port or a communication port.
 
-How to Configure a Console Port
-===============================
+vUART enables you to exchange data between the hypervisor and a VM
```
Comment (dbkinder) [inline doc/tutorials/vuart_configuration.rst:12]:
```suggestion
A vUART can exchange data between the hypervisor and a VM
```

File: doc/tutorials/vuart_configuration.rst [hunk 2]
```diff
@@ -3,386 +3,151 @@
...
+or between two VMs. Typical use cases of vUART include:
```
Comment (dbkinder) [inline doc/tutorials/vuart_configuration.rst:13]:
```suggestion
or between two VMs. Typical use cases of a vUART include:
```

File: doc/tutorials/vuart_configuration.rst [hunk 3]
```diff
@@ -3,386 +3,151 @@
...
-- ``COM3_BASE (0x3E8) + COM3_IRQ(6)``
-- ``COM4_BASE (0x2E8) + COM4_IRQ(7)``
+* Exchange user-specific, low-speed data between two VMs.
 
-Example:
+To the VMs, the vUARTs are presented in a 8250-compatible manner.
 
-.. code-block:: none
+To exchange high-speed (for example, megabytes or gigabytes per second) data
+between two VMs, you can use the shared memory feature (by means of QEMU's
```
Comment (dbkinder) [inline doc/tutorials/vuart_configuration.rst:24]:
```suggestion
between two VMs, you can use the inter-VM shared memory feature
```

File: doc/tutorials/vuart_configuration.rst [hunk 4]
```diff
@@ -3,386 +3,151 @@
...
+IVSHMEM) instead.
```
Comment (dbkinder) [inline doc/tutorials/vuart_configuration.rst:25]:
```suggestion
(IVSHMEM) instead.
```

File: doc/tutorials/vuart_configuration.rst [hunk 5]
```diff
@@ -3,386 +3,151 @@
...
 
-How to Configure a Communication Port
-=====================================
+* The OSes of the VMs need an 8250-compatible serial driver.
 
-To enable the communication port, configure ``vuart[1]`` in the two VMs that want to communicate.
+* To access the hypervisor shell, you must have a physical UART.
 
-The port_base and IRQ should differ from the ``vuart[0]`` in the same VM.
+* Although vUART is available to all kinds of VMs, it is not recommended to
```
Comment (dbkinder) [inline doc/tutorials/vuart_configuration.rst:36]:
```suggestion
* Although a vUART is available to all kinds of VMs, you should not
```

File: doc/tutorials/vuart_configuration.rst [hunk 6]
```diff
@@ -3,386 +3,151 @@
...
+  enable vUART to access the console of or exchange data with a VM that needs to
```
Comment (dbkinder) [inline doc/tutorials/vuart_configuration.rst:37]:
```suggestion
  enable a vUART to access the console of or exchange data with a real-time VM.
```

File: doc/tutorials/vuart_configuration.rst [hunk 7]
```diff
@@ -3,386 +3,151 @@
...
+  meet real-time requirements. Exchanging data via vUART imposes a performance
+  penalty that could delay the response of asynchronous events in those VMs.
```
Comment (dbkinder) [inline doc/tutorials/vuart_configuration.rst:39]:
```suggestion
  penalty that could delay the response of asynchronous events in real-time VMs.
```

File: doc/tutorials/vuart_configuration.rst [hunk 8]
```diff
@@ -3,386 +3,151 @@
...
+.. image:: images/configurator-vuartconn01.png
+   :align: center
+   :class: drop-shadow
 
-   On VM2: ``# echo "test test" > /dev/ttyS1``
+Example Configuration
+=====================
 
-   You can find the message from VM1 ``/dev/ttyS1``.
+The following steps show an example of how to configure and verify a vUART
```
Comment (dbkinder) [inline doc/tutorials/vuart_configuration.rst:88]:
```suggestion
The following steps show how to configure and verify a vUART
```

**Review by dbkinder — CHANGES_REQUESTED** [review]

**Review by dbkinder — APPROVED** [review]
LGTM, thanks

**Pull request merged**